core/src/main/resources/error/error-classes.json

```diff
@@ -912,6 +912,11 @@
     ],
     "sqlState" : "22023"
   },
+  "STAR_GROUP_BY_POS" : {
```

Comment:
LGTM ... but why do we have this restriction?

Reply:
It is better to ask the author of the restriction, cc @gatorsmile 
https://github.com/apache/spark/pull/11846/files#diff-ed19f376a63eba52eea59ca71f3355d4495fad4fad4db9a3324aade0d4986a47R389-R392